src/stores/AppStore.ts

```diff
@@ -0,0 +1,34 @@
+import { makeAutoObservable } from "mobx";
+import { CityWithUTCOffset } from 'types';
+
+const now = new Date();
+const baseTime = now.getTime() + now.getTimezoneOffset() * 60 * 1000; // no utc offset
+
+export const appStore = makeAutoObservable({
```

Comment:
I suggest you try a class-based approach, with a singleton pattern via [tsyringe library](https://github.com/microsoft/tsyringe#singleton)

```
@singleton()
class CityListStore {
  constructor() {
     makeAutoObservable()
  }
}

export const useCityListStore = () => container.resolve(CityListStore)
```


Using this, we can easily create atomic stores and we don't need to use react context boilerplate. 

In this case, ideally, you should create two stores CityListStore and TimerStore and incapsulate all logic related to search functionality / timer updates here